### DISCUSSION TITLE
mesh and xml file for Kuka Quantec KR180 R2500

### DISCUSSION THREAD

**Discussion by aimforkim** [visose/Robots#216] [Q&A] [answered]
Hi,
I would like to use this plugin to simulate and generate toolpath for the Kuka Quantec KR180 R2500. Is there a xml file and mesh file of this model available? 

I haven´t been able to find the xml file for the Quantec series so far. Any help would be hugely appreciated!

**Comment by visose** [comment]
You can search if it exists in this branch here: https://github.com/visose/robots/tree/libraries

**Reply by aimforkim** [reply]
Hi, thank you for the reply. 

I scrolled through all the xml files in this library, and I could not find the KR180-R2500. I did try to write my own xml file by referencing the existing kuka kr6 and kr500 xml. The link below is my Kuka Kr180 mesh model, and the xml file that I tried to write. 

I placed it in the documents folder with my other robot files, and I got this error: "The provided path is null, does not exist or cannot be accessed." What could I be missing?

![image](https://user-images.githubusercontent.com/96722124/216948026-33ebf31a-7606-4eb6-90e5-faebd1628599.png)

https://drive.google.com/drive/folders/17TuKBZuFnb6WGg2QwSb8dshnepTmV6IL?usp=share_link

**Comment by aimforkim** [comment, marked as answer]
Hi!
I managed to resolve this. I believe it is because of the way I saved the Rhino model of the mesh, potentially also the way I named the layer.
Two changes I've made in Rhino: 
1. Layer name: RobotArm.KUKA.<name of the robot model>
2. I made sure that each layer of the axis (0 to 6) contains only one joined mesh. 

With this, I will close the issue.
